cmd/argo/commands/submit.go

```diff
@@ -92,6 +92,47 @@ func SubmitWorkflows(filePaths []string, submitOpts *util.SubmitOpts, cliOpts *c
 		workflows = append(workflows, wfs...)
 	}
 
+	submitWorkflows(workflows, submitOpts, cliOpts)
+}
+
+func SubmitWorkflowFromResource(resourceIdentifier string, submitOpts *util.SubmitOpts, cliOpts *cliSubmitOpts) {
+
+	resIdSplit := strings.Split(resourceIdentifier, "/")
+	if len(resIdSplit) != 2 {
+		log.Fatalf("resource identifier '%s' is malformed. Expected is KIND/NAME, e.g. CronWorkflow/hello-world-cwf", resourceIdentifier)
+	}
+
+	var workflowToSubmit *wfv1.Workflow
+	switch resIdSplit[0] {
+	case workflow.CronWorkflowKind:
+		cwfIf := cron.InitCronWorkflowClient()
+		cwf, err := cwfIf.Get(resIdSplit[1], v1.GetOptions{})
+		if err != nil {
+			log.Fatalf("Unable to get CronWorkflow '%s': %s", resIdSplit[1], err)
+		}
+		workflowToSubmit, err = common.ConvertCronWorkflowToWorkflow(cwf)
+		if err != nil {
+			log.Fatalf("Unable to create Workflow from CronWorkflow '%s': %s", resIdSplit[1], err)
+		}
+	default:
+		log.Fatalf("Resource Kind '%s' is not supported with --from", resIdSplit[0])
+	}
+
+	submitWorkflows([]wfv1.Workflow{*workflowToSubmit}, submitOpts, cliOpts)
+}
+
+func submitWorkflows(workflows []wfv1.Workflow, submitOpts *util.SubmitOpts, cliOpts *cliSubmitOpts) {
+	if submitOpts == nil {
+		submitOpts = &util.SubmitOpts{}
+	}
+	if cliOpts == nil {
+		cliOpts = &cliSubmitOpts{}
+	}
+
+	ctx, apiClient := client.NewAPIClient()
```

Comment:
I think we should only create one of these per CLI command invocation, but I struggle with how to pass it between functions. I wonder if we could create a struct this those fields and invoke on this

```
type SubmitCommand struct {
cobra.Command
}

func (c SubmitCommand) submitWorkflows() {...}
```